sklearn/utils/arrayfuncs.pyx

```diff
@@ -24,6 +24,25 @@ def min_pos(const floating[:] X):
     return min_val
 
 
+def any_zero_row(floating[:, :] X):
```

Comment:
I would make this function private for the moment. I think that we can make the function slightly more generatic and looking like NumPy:

```python
ctypedef fused real_numeric:
    short
    int
    long
    float
    double


def _all_with_any_reduction(real_numeric[:, :] array, real_numeric value=0, int axis=0):
    """Check all values are equal to `value` along a specific axis.

    It is equivalent to `np.any(np.all(X == value, axis=axis))` but it avoids to
    materialize the temporary boolean matrices in memory.

    Parameters
    ----------
    array : array-like
        The array to be checked.
    value : short, int, long, float, double
        The value to use for the comparison.
    axis : int, default=0
        Along which axis to perform the `all` operation.

    Returns
    -------
    bool
        Whether or not any rows (i.e. `axis=1`) or any columns (i.e. `axis=0`) contains
        all values equal to `value`.
    """
    cdef:
        Py_ssize_t i, j

    if axis == 0:
        for j in range(X.shape[1]):
            for i in range(X.shape[0]):
                if X[i, j] != value:
                    break
            else:
                return True
        return False
    else:  # axis == 1:
        for i in range(X.shape[0]):
            for j in range(X.shape[1]):
                if X[i, j] != value:
                    break
            else:
                return True
        return False
```